docs/language/learn-ql/csharp/basic-query-csharp.rst

```diff
@@ -0,0 +1,145 @@
+|                                                               |                                                                                                                   | ``block.isEmpty()`` states that the block must be empty (that is, it contains no statements).                          |
++---------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+| ``select ifstmt, "This 'if' statement is redundant."``        | Defines what to report for each match.                                                                            | Reports the resulting ``if`` statement with a string that explains the problem.                                        |
+|                                                               |                                                                                                                   |                                                                                                                        |
+|                                                               | ``select`` statements for queries that are used to find instances of poor coding practice are always in the form: |                                                                                                                        |
+|                                                               | ``select <program element>, "<alert message>"``                                                                   |                                                                                                                        |
++---------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------+
+
+Extend the query
+----------------
+
+Query writing is an inherently iterative process. You write a simple query and then, when you run it, you discover examples that you had not previously considered, or opportunities for improvement.
+
+Remove false positive results
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Browsing the results of our basic query shows that it could be improved. Among the results you are likely to find examples of ``if`` statements with an ``else`` branch, where an empty ``then`` branch does serve a purpose. For example:
+
+.. code-block:: csharp
+
+   if (...) {
+     ...
+   } else if (string.Compare(option, "-verbose")==0) {
```

Comment:
I am not familiar with C#, but it appears you would normally use `Equals` here?
Changing that would make this example more relatable.

Reply:
Thanks for bringing this up! I've updated the example to use more common syntax (and indentation).